codecov/gazebo
# feat: new UI to clarify adding token

# Description

Closes https://github.com/codecov/engineering-team/issues/2698

Figma: https://www.figma.com/design/ZJmSdJUIz03zJdF22kLhAs/GH-2694?node-id=2058-3163&t=koQcNbcBaHcFN7rk-1 There are two versions in the file. Check the higher up version that is labeled "Latest".

With the addition of org level tokens and token auth possibly being required for an org, we have multiple variations as outlined in the figma file.

Landing on this repo onboarding page...
- the org token may or may not have been previously generated (could have been generated in their settings)
- the org may or may not require token authentication (this is set in their settings)
- the user may be an admin or just a regular member

Easiest way to test on preview:
If you have any repos on your personal org, go to the onboarding for one.
- Check what the steps look like without org token generated for both "token required" and "not token required" (setting can be changed in in the "Settings" tab -> Global Upload Token on org level.
- Click "Generate" on onboarding step and check what the steps look like for both token required/not required

# Notable Changes

I changed the Step components to move away from naming with number specific names and I also split them out from GitHubActions.tsx as that file was getting quite large.

# Screenshots

I've only expanded the screenshot example for "Token required - Prev Generated" but all of the others would look the same.

Token required - Not Prev Generated
<img width="829" alt="Required - No Prev" src="https://github.com/user-attachments/assets/274b4872-ef4f-40b3-bb1b-decb34feac92" />
Token required - Not Prev Generated but newly generated in this flow
<img width="881" alt="Required - Generated" src="https://github.com/user-attachments/assets/330593dd-fa97-4b5e-8519-f4075fdf0d3f" />
Token required - Repo instruction chosen (Not Prev Generated)
<img width="864" alt="Required - Repo Chosen" src="https://github.com/user-attachments/assets/af5d626f-49c8-4df7-9df4-e163d281b072" />
Token required - Prev Generated
<img width="1013" alt="Required - Prev" src="https://github.com/user-attachments/assets/51d9bf58-cc77-4c22-9803-17c14150059e" />

Token optional - Not Prev Generated
<img width="829" alt="Optional - No Prev" src="https://github.com/user-attachments/assets/dddab730-1c75-4e3d-9edf-16984d9087c6" />
Token optional - Prev Generated
<img width="916" alt="Optional - Prev" src="https://github.com/user-attachments/assets/4f7fbaac-ba6a-435b-b6fe-b26383b203f2" />
Token optional - Not Prev Generated but newly generated in this flow
<img width="896" alt="Optional - Generated" src="https://github.com/user-attachments/assets/9e53f9ff-ad96-47ba-8c2e-e134af44ecb4" />

Non-admin Disabled Generated Button
<img width="826" alt="NonAdmin" src="https://github.com/user-attachments/assets/dc9e64e2-f2d1-4991-8408-fdeeb694afb1" />

Light mode screenshot
![Screenshot 2024-12-12 at 3 12 17 PM](https://github.com/user-attachments/assets/b4f21dd0-e50f-4da1-a02c-8071d0a6d809)



# Link to Sample Entry



# Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. In 2022 this entity acquired Codecov and as result Sentry is going to need some rights from me in order to utilize my contributions in this PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.
